qiskit/transpiler/instruction_durations.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Durations of instructions, one of transpiler configurations."""
+import warnings
+from typing import Optional, List, Tuple, Union, Iterable
+
+from qiskit.circuit import Barrier, Delay
+from qiskit.circuit import Instruction, Qubit
+from qiskit.providers import BaseBackend
+from qiskit.transpiler.exceptions import TranspilerError
+
+
+class InstructionDurations:
+    """Helper class to provide integer durations for safe scheduling."""
+
+    def __init__(self,
+                 instruction_durations: Optional['InstructionDurationsType'] = None,
+                 schedule_dt=None):
+        self.duration_by_name = {}
+        self.duration_by_name_qubits = {}
+        self.schedule_dt = schedule_dt
+        if instruction_durations:
+            self.update(instruction_durations, schedule_dt)
+
+    @classmethod
+    def from_backend(cls, backend: BaseBackend):
+        """Construct the instruction durations from the backend."""
+        if backend is None:
+            return InstructionDurations()
+        # TODO: backend.properties() should tell us all about instruction durations
+        if not backend.configuration().open_pulse:
+            raise TranspilerError("No backend.configuration().dt in the backend")
```

Comment:
You construct instruction durations from backend properties, so why error if they don't have `dt` (are not pulse enabled)?